### PR TITLE
Meta: Implement support for the new String in the gdb formatter

### PR DESCRIPTION
**Pull request by SebastianZaha**
The pretty-print gdb helpers were not updated since the DeprecatedString change. This commit introduces a new printer for String and renames the old one to DeprecatedString.

StringImpl and StringView were also displayed character-by-character in the pretty printer. This is now changed to properly show the full "concatenated" string.

### CODE REVIEW

File: Meta/serenity_gdb.py [hunk 1]
```diff
@@ -169,9 +189,7 @@ def to_string(self):
         if int(self.val["m_length"]) == 0:
             return '""'
         else:
-            characters = self.val["m_characters"]
-            str_type = characters.type.target().array(self.val["m_length"]).pointer()
-            return str(characters.cast(str_type).dereference())
+            return self.val["m_characters"].string(length=self.val["m_length"])
```
Comment (LucasChollet) [inline Meta/serenity_gdb.py:192]: I think that the "read the string as a string" part should be in its own commit

Comment (SebastianZaha) [reply Meta/serenity_gdb.py:192]: You're right, looks cleaner this way. I reworded the commit message too, it was a bit awkwardly phrased.

**Review by LucasChollet — APPROVED** [review]

**Comment by alimpfard** [comment]
Thanks for fixing this!
